src/services/operator-value-parser.js

```diff
@@ -1,81 +0,0 @@
-import _ from 'lodash';
-import { Schemas } from 'forest-express';
-import Operators from '../utils/operators';
-import OperatorDateIntervalParser from './operator-date-interval-parser';
-
-function OperatorValueParser(options) {
-  const OPERATORS = new Operators(options);
-
-  this.perform = (model, fieldName, value, timezone) => {
-    const operatorDateIntervalParser = new OperatorDateIntervalParser(
-      value,
-      timezone,
-      options,
-    );
-
-    // NOTICE: Handle boolean for MySQL database
-    let modelName;
-    let field;
-    let fieldSplit;
-    let valueBoolean;
-    let fieldBoolean = false;
-
-    if (fieldName.indexOf(':') === -1) {
-      modelName = model.name;
-    } else {
-      fieldSplit = fieldName.split(':');
-      modelName = fieldSplit[0];
-      fieldName = fieldSplit[1];
-    }
-
-    const schema = Schemas.schemas[modelName];
-    if (schema) {
-      field = _.find(schema.fields, currentField => currentField.field === fieldName);
-      if (field && field.type === 'Boolean') {
-        fieldBoolean = true;
-        if (value.indexOf('true') > -1) {
-          valueBoolean = true;
-        } else if (value.indexOf('false') > -1) {
-          valueBoolean = false;
-        }
-      }
-    }
-
-    const condition = {};
-    if (value[0] === '!' && value[1] !== '*') {
-      value = value.substring(1);
-      if (fieldBoolean) {
-        condition[OPERATORS.NOT] = _.isUndefined(valueBoolean) ? null : valueBoolean;
```

Comment:
Not convinced boolean are still well handled with the new implementation.
An integration test (if there is not) could secure this from a regression.

Reply:
I don't really get what kind of integration test I should do.